adapter/src/dbt/adapters/fal_experimental/utils.py

```diff
@@ -80,3 +84,54 @@ def db_adapter_config(config: RuntimeConfig) -> RuntimeConfig:
         new_config = config
 
     return new_config
+
+
+def load_environments(base_dir: str) -> Dict[str, "BaseEnvironment"]:
+    import os
+    fal_project_path = os.path.join(base_dir, "fal_project.yml")
+    if not os.path.exists(fal_project_path):
+        raise FalParseError(f"{fal_project_path} must exist to define environments")
+
+    fal_project = load_yaml(fal_project_path)
+    environments = {}
+    for environment in fal_project.get("environments", []):
+        env_name = _get_required_key(environment, "name")
+        if _is_local_environment(env_name):
+            raise FalParseError(
+                f"Environment name conflicts with a reserved name: {env_name}."
+            )
+
+        env_kind = _get_required_key(environment, "type")
+        environments[env_name] = create_environment(env_name, env_kind, environment)
+    return environments
+
+
+def create_environment(name: str, kind: str, config: Dict[str, Any]):
+    from isolate.backends.virtualenv import VirtualPythonEnvironment
+    from isolate.backends.conda import CondaEnvironment
+
+    REGISTERED_ENVIRONMENTS: Dict[str, BaseEnvironment] = {
+        "conda": CondaEnvironment,
+        "venv": VirtualPythonEnvironment,
+    }
+    env_type = REGISTERED_ENVIRONMENTS.get(kind)
+    if env_type is None:
+        raise ValueError(
+            f"Invalid environment type (of {kind}) for {name}. Please choose from: "
+            + ", ".join(REGISTERED_ENVIRONMENTS.keys())
+        )
+
+    parsed_config = {
+        'requirements': config.get('requirements', [])
```

Comment:
`packages` for conda?

Reply:
Maybe `requirements` are fine? @isidentical, do you have an opinion on this?

Reply:
For conda, both the `CondaEnvironment` and the existing fal uses `packages`. I think we have changed it after some discussion so we could stick with it. No strong opinions though.